scripts/Expect.py

```diff
@@ -731,3 +732,41 @@ def test(result, expected):
                 test(self.exitstatus, [exitstatus, -self.killed])
         else:
             test(self.exitstatus, exitstatus)
+
+    def stackDump(self):
+        match platform.system():
+            case "Linux":
+                cmd = [
+                    "gdb",
+                    "-q",
+                    "-n",
+                    "-batch",
+                    "-ex",
+                    f"attach {self.p.pid}",
+                    "-ex",
+                    "thread apply all bt",
+                    "-ex",
+                    "detach",
+                    "-ex",
+                    "quit",
+                ]
+            case "Darwin":
+                cmd = [
+                    "lldb",
+                    "-p",
+                    f"{self.p.pid}",
+                    "-o",
+                    "thread backtrace all",
+                    "-o",
+                    "process detach",
+                    "-o",
+                    "quit",
+                ]
+            case _:
+                print(f"stackDump not supported on {platform.system()}")
+                return
+
+        try:
+            subprocess.run(cmd)
```

Comment:
It would be helpful to log a message like `Process xxx is hanging, dumping all threads...` before running gdb, just to make it more obvious what’s happening.

Also, do we run this from the main thread? If so, it might be better to redirect the output to a file and upload it as a CI artifact.